lib/googleauth/signet.rb

```diff
@@ -89,6 +93,17 @@ def retry_with_error max_retry_count = 5
           end
         end
       end
+
+      private
+
+      def expires_at_from_id_token id_token
+        match = /^[\w=-]+\.([\w=-]+)\.[\w=-]+$/.match id_token.to_s
+        return unless match
+        base64 = Base64.urlsafe_decode64 match[1]
+        json = JSON.parse base64 rescue nil
+        return unless json.respond_to?(:key?) && json.key?("exp")
+        Time.at json["exp"].to_i
```

Comment:
can this fail / does it need a rescue?

Reply:
I don't think so, but good idea to wrap it anyway for safety.